### PR TITLE
71098 merge

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**

### CODE REVIEW

File: src/EA.Weee.Core/Organisations/RepresentingCompanyDetailsViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        public string OrganisationType { get; set; }
+
+        [Required]
+        [DisplayName("Producer name")]
+        [StringLength(CommonMaxFieldLengths.DefaultString)]
+        public string CompanyName { get; set; }
+
+        [Required]
+        [DisplayName("Trading name")]
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Core/Organisations/RepresentingCompanyDetailsViewModel.cs:19]: This isnt required.

File: src/EA.Weee.Core/Organisations/RepresentingCompanyDetailsViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        public string BusinessTradingName { get; set; }
+
+        public AddressData Address { get; set; } = new AddressData();
+
+        [StringLength(CommonMaxFieldLengths.Telephone)]
+        [Display(Name = "Telephone number")]
+        [GenericPhoneNumber(ErrorMessage = "The telephone number can use numbers, spaces and some special characters (-+). It must be no longer than 20 characters.")]
+        public string Telephone { get; set; }
+
+        [Display(Name = "Email address")]
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Core/Organisations/RepresentingCompanyDetailsViewModel.cs:29]: This is required.

File: src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+﻿@using EA.Weee.Core.DataStandards
+@using EA.Weee.Web.Areas.Admin.Controllers
+@using MvcNavigationHelpers
+@using EA.Weee.Web.Controllers
+@model EA.Weee.Core.Organisations.RepresentingCompanyDetailsViewModel
+@{
+    ViewBag.Title = "Producer details";
+}
+@section main_content
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml:9]: Upadte to use full_main_content.

File: src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+{
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml:10]: Need a back link.

File: src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+        <div class="govuk-form-group @Html.Gds().FormGroupClass(m => m.Address)">
+            @Html.EditorFor(m => m.Address)
+        </div>
+
+        <div class="govuk-form-group">
+            <button class="govuk-button" type="submit" data-module="govuk-button" data-prevent-double-click="true">Continue</button>
+        </div>
+        <p>
+            <a href="@(Url.UrlFor<OrganisationRegistrationController>(u => u.Type()))">Back</a>
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml:47]: Move to top.

File: src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        </p>
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml:48]: Need a back to top link. Use another view as example.

File: src/EA.Weee.Web/Services/OrganisationTransactionService.cs [hunk 7]
```diff
@@ -37,6 +37,12 @@ public async Task CaptureData<T>(string accessToken, T model) where T : class
                     case PreviousRegistrationViewModel previousRegistrationModel:
                         transaction.PreviousRegistration = previousRegistrationModel.SelectedValue.GetValueFromDisplayName<YesNoType>();
                         break;
+                    case RepresentingCompanyDetailsViewModel representingCompanyDetailsViewModel:
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Services/OrganisationTransactionService.cs:40]: Dont clear the other fields here as they are used by other pages,

**Comment by JohnPadgettCivica** [comment]
@stevelee-civica I have create a mergePR which I think resolves issues with the merge, also added comments.

File: src/EA.Weee.Core/Organisations/RepresentingCompanyDetailsViewModel.cs (hunk 1 above)
Comment (stevelee-civica) [reply src/EA.Weee.Core/Organisations/RepresentingCompanyDetailsViewModel.cs:19]: updated

File: src/EA.Weee.Core/Organisations/RepresentingCompanyDetailsViewModel.cs (hunk 2 above)
Comment (stevelee-civica) [reply src/EA.Weee.Core/Organisations/RepresentingCompanyDetailsViewModel.cs:29]: updated

File: src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml (hunk 3 above)
Comment (stevelee-civica) [reply src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml:9]: updated

File: src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml (hunk 4 above)
Comment (stevelee-civica) [reply src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml:10]: updated

File: src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml (hunk 5 above)
Comment (stevelee-civica) [reply src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml:47]: updated

File: src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        <div>
+            <h2 class="govuk-heading-s">Contact address</h2>
+        </div>
+
+        <div class="govuk-form-group @Html.Gds().FormGroupClass(m => m.Address)">
+            @Html.EditorFor(m => m.Address)
+        </div>
+
+        <div class="govuk-form-group">
+            <button class="govuk-button" type="submit" data-module="govuk-button" data-prevent-double-click="true">Continue</button>
```
[the remark this answers is not in the document]
Comment (stevelee-civica) [reply src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml:44]: updated

File: src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml (hunk 6 above)
Comment (stevelee-civica) [reply src/EA.Weee.Web/Views/OrganisationRegistration/RepresentingCompanyDetails.cshtml:48]: updated

File: src/EA.Weee.Web/Services/OrganisationTransactionService.cs (hunk 7 above)
Comment (stevelee-civica) [reply src/EA.Weee.Web/Services/OrganisationTransactionService.cs:40]: updated

File: src/EA.Weee.Web/Controllers/OrganisationRegistrationController.cs [hunk 9]
```diff
@@ -289,6 +289,50 @@ public async Task<ActionResult> Type(OrganisationTypeViewModel model)
...
+            {
+                var countries = await GetCountries();
+
+                model.Address.Countries = countries;
+
+                return View(model);
+            }
+
+            await transactionService.CaptureData(User.GetAccessToken(), model);
+
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Controllers/OrganisationRegistrationController.cs:306]: @stevelee-civica Noticed that here we also need to call 
await transactionService.CompleteTransaction(User.GetAccessToken());
await cache.InvalidateOrganisationSearch();

As we need to mark the organisation transaction as complete. Could you add that after capture data and add the relevant unit tests similar to how others are done.

**Pull request closed without merging**